PowerShell/ScubaGear/Testing/Unit/Rego/Sharepoint/SharepointConfig_03_test.rego

```diff
@@ -153,73 +153,146 @@ test_SharingCapability_NewExistingGuests_NotApplicable_V2 if {
-                {"op": "add", "path": "FileAnonymousLinkType", "value": 2}])
+    # Set PnP flag to true denoting use of service principal
+    Output := sharepoint.tests with input.SPO_tenant as [Tenant]
+        with input.OneDrive_PnP_Flag as true
+    TestResult("MS.SHAREPOINT.3.2v1", Output, PASS, true) == true
+}
 
+test_File_Folder_AnonymousLinkType_UsingServicePrincipal_Incorrect if {
+    # SharingCapability value of 2 equals "Anyone"
+    # FileAnonymousLinkType value of 2 equals "Edit"
+    # FolderAnonymousLinkType value of 2 equals "Edit"
+    Tenant := json.patch(SPOTenant, [
+        {"op": "add", "path": "SharingCapability", "value": 2},
+        {"op": "add", "path": "FileAnonymousLinkType", "value": 2},
+        {"op": "add", "path": "FolderAnonymousLinkType", "value": 2}
+    ])
+    
+    # Set PnP flag to true denoting use of service principal 
+    Output := sharepoint.tests with input.SPO_tenant as [Tenant]
+        with input.OneDrive_PnP_Flag as true
+
+    ReportDetailsString := concat(": ", [
+        FAIL,
+        "both files and folders are not limited to view for Anyone"
+    ])
+    TestResult("MS.SHAREPOINT.3.2v1", Output, ReportDetailsString, false) == true
+}
+
+test_File_AnonymousLinkType_UsingServicePrincipal_Incorrect if {
+    # SharingCapability value of 2 equals "Anyone"
+    # FileAnonymousLinkType value of 2 equals "Edit"
+    # FolderAnonymousLinkType value of 1 equals "View"
+    Tenant := json.patch(SPOTenant, [
+        {"op": "add", "path": "SharingCapability", "value": 2},
+        {"op": "add", "path": "FileAnonymousLinkType", "value": 2},
+        {"op": "add", "path": "FolderAnonymousLinkType", "value": 1}
+    ])
+    
+    # Set PnP flag to true denoting use of service principal
+    Output := sharepoint.tests with input.SPO_tenant as [Tenant]
+        with input.OneDrive_PnP_Flag as true
+
+    # FAIL = Requirement not met
+    # ReportDetailsString = "Requirement not met: both files and folders are not limited to view for Anyone"
+    ReportDetailsString := concat(": ", [
+        FAIL,
+        "files are not limited to view for Anyone"
+    ])
+    TestResult("MS.SHAREPOINT.3.2v1", Output, ReportDetailsString, false) == true
+}
+
+test_Folder_AnonymousLinkType_UsingServicePrincipal_Incorrect if {
+    # SharingCapability value of 2 equals "Anyone"
+    # FileAnonymousLinkType value of 1 equals "View"
+    # FolderAnonymousLinkType value of 2 equals "Edit"
+    Tenant := json.patch(SPOTenant, [
+        {"op": "add", "path": "SharingCapability", "value": 2},
+        {"op": "add", "path": "FileAnonymousLinkType", "value": 1},
+        {"op": "add", "path": "FolderAnonymousLinkType", "value": 2}
+    ])
+    
+    # Set PnP flag to true denoting use of service principal
     Output := sharepoint.tests with input.SPO_tenant as [Tenant]
-                                with input.OneDrive_PnP_Flag as true
+        with input.OneDrive_PnP_Flag as true
```

Comment:
I thought we were removing this?

Reply:
It's only removed in the SharePoint.rego file. When running Invoke-SCuBA with the AppId parameter this sets `$PnPFlag = $true` in Orchestrator.ps1.

![Screenshot (230)](https://github.com/user-attachments/assets/d6d742b0-3c25-473b-bdef-28e8fbe92a4c)

It's then passed into the SharePoint provider to conditionally run either Get-PnPTenant/Get-PnpTenantSite or Get-SPOTenant/Get-SPOSite commandlets. 

![Screenshot (232)](https://github.com/user-attachments/assets/70e2deb1-1459-4be5-857e-9a29c3611883)

The purpose of keeping `input.OneDrive_PnP_Flag as true` in the unit tests is to mock running Invoke-SCuBA via service principal, otherwise the tests will default to the interactive authentication variant.